.github/workflows/tf-deploy-basic.yaml

```diff
@@ -46,73 +68,138 @@ jobs:
+      - name: Terraform Plan
+        id: plan
         run: |
+          # Terraform Pan Script
+          # As this script grows, it may be worth moving into its own file.
+          #
+          # This script is responsible for running the terraform plan command and saving the output to a file.
+          # Exit codes:
+          # 0 - No changes
+          # 1 - Error
+          # 2 - Changes
+          #
+          # TODO: State lock is required if plan is being saved to a file. Are all backends supported?
+
+          export exitcode=0
+
+          echo "Current working directory: $(pwd)"
+
           tf_args=''
-          if [ "${{ inputs.var_file }}" == false ]; then
-            tf_args='-var-file="environments/${{ matrix.environment }}.tfvars"'
-          fi
+
+          IFS=',' read -ra ADDR <<< "${{ inputs.tf_vars }}"
+          for var in "${ADDR[@]}"; do
+            tf_args+=" -var $var"
+          done
+
+          IFS=',' read -ra ADDR <<< "${{ inputs.tf_var_files }}"
+          for var_file in "${ADDR[@]}"; do
+            tf_args+=" -var-file=$var_file"
+          done
 
           terraform plan \
+           -detailed-exitcode \
+           -no-color \
            -input=false \
            -lock=false \
            -var="environment=${{ inputs.environment }}" \
            -var="github_repo=${{ github.event.repository.name }}" \
+           -var="region=${{ inputs.aws_region }}" \
            -out="plan" \
-           ${tf_args}
+           ${tf_args} || exitcode=$?
+
+          echo "exitcode=$exitcode" >> $GITHUB_OUTPUT
+
+          if [ $exitcode -eq 1 ]; then
+            echo Terraform Plan Failed!
+            exit 1
+          else
+            exit 0
+          fi
+
+      - name: Create String Output
+        id: tf-plan-string
+        run: |
+          TERRAFORM_PLAN=$(terraform show -no-color plan)
+
+          delimiter="$(openssl rand -hex 8)"
+          echo "summary<<${delimiter}" >> $GITHUB_OUTPUT
+          echo "# Terraform Plan Output: ${{ inputs.environment }}" >> $GITHUB_OUTPUT
+          echo "<details><summary>Click to expand</summary>" >> $GITHUB_OUTPUT
+          echo "" >> $GITHUB_OUTPUT
+          echo '```terraform' >> $GITHUB_OUTPUT
+          echo "$TERRAFORM_PLAN" >> $GITHUB_OUTPUT
+          echo '```' >> $GITHUB_OUTPUT
+          echo "</details>" >> $GITHUB_OUTPUT
+          echo "${delimiter}" >> $GITHUB_OUTPUT
+
+      - name: Publish Terraform Plan to Task Summary
+        env:
+          SUMMARY: ${{ steps.tf-plan-string.outputs.summary }}
+        run: |
+          echo "$SUMMARY" >> $GITHUB_STEP_SUMMARY
 
       - name: Terraform Apply
-        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push' && steps.plan.outputs.exitcode == '2' || inputs.tf_deploy_override == true # https://developer.hashicorp.com/terraform/cli/commands/plan#detailed-exitcode
```

Comment:
we should pass the "main" or "master" parameter no ? or use a magical git command to get the default branch ? found this:
```
git remote show origin | sed -n '/HEAD branch/s/.*: //p' 
````

Reply:
((github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master') && github.event_name == 'push' && steps.plan.outputs.exitcode == '2') || inputs.tf_deploy_override == true

😶‍🌫️